# Description: some sessions have been renamed between Natty and Oneiric.
#              Make the transition happens as less intrusive as possible.
=== modified file 'liblightdm-gobject/user.c'
Index: ubuntu/liblightdm-gobject/user.c
===================================================================
--- ubuntu.orig/liblightdm-gobject/user.c	2011-09-06 16:10:26.454205000 +0200
+++ ubuntu/liblightdm-gobject/user.c	2011-09-06 16:12:46.528048812 +0200
@@ -1105,6 +1105,22 @@
     priv->language = g_key_file_get_string (priv->dmrc_file, "Desktop", "Language", NULL);
     priv->layout = g_key_file_get_string (priv->dmrc_file, "Desktop", "Layout", NULL);
     priv->session = g_key_file_get_string (priv->dmrc_file, "Desktop", "Session", NULL);
+
+    if (g_strcmp0 (priv->session, "gnome") == 0)
+    {
+        g_free(priv->session);
+        priv->session = g_strdup ("ubuntu");
+    }
+    else if (g_strcmp0 (priv->session, "unity-2d") == 0)
+    {
+        g_free(priv->session);
+        priv->session = g_strdup ("ubuntu-2d");
+    }
+    else if (g_strcmp0 (priv->session, "gnome-2d") == 0)
+    {
+        g_free(priv->session);
+        priv->session = g_strdup ("gnome-fallback");
+    }
 }
 
 static gchar *
